.github/workflows/publish-new-release.yml

```diff
@@ -0,0 +1,191 @@
+          VERSION=${BRANCH_NAME#release/}
+
+          echo "::set-output name=version::$VERSION"
+
+      - name: Publish docker image as ${{ steps.extract-version.outputs.version }} and latest
+        run: |
+          VERSION="${{ steps.extract-version.outputs.version }}"
+
+          docker build . -t comitnetwork/cnd:$VERSION -t comitnetwork/cnd:latest
+          docker push comitnetwork/cnd:$VERSION
+          docker push comitnetwork/cnd:latest
+
+
+  release:
+    if: github.event.pull_request.merged == true && startsWith(github.event.pull_request.head.ref, 'release/') # only merged release branches must trigger this
+    name: Create GitHub release
+    needs: build_binary
+    runs-on: ubuntu-latest
+    steps:
+      - name: Extract version from branch name
+        id: extract-version
+        run: |
+          BRANCH_NAME="${{ github.event.pull_request.head.ref }}"
+          VERSION=${BRANCH_NAME#release/}
+
+          echo "::set-output name=version::$VERSION"
+
+      - name: Download ubuntu release package
+        uses: actions/download-artifact@v1
+        with:
+          name: ubuntu-release-archive
+
+      - name: Download macos release package
+        uses: actions/download-artifact@v1
+        with:
+          name: macos-release-archive
+
+      - name: Detect archives to upload
+        id: detect-archives
+        run: |
+          echo "::set-output name=ubuntu-archive::$(cd ./ubuntu-release-archive/; echo *.tar.gz)"
+          echo "::set-output name=macos-archive::$(cd ./macos-release-archive/; echo *.tar.gz)"
+
+      - name: Create Release
+        id: create-release
+        uses: thomaseizinger/create-release@1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          target_commitish: ${{ github.event.pull_request.merge_commit_sha }}
+          tag_name: ${{ steps.extract-version.outputs.version }}
+          name: ${{ steps.extract-version.outputs.version }}
+          draft: false
+          prerelease: false
+
+      - name: Upload ubuntu release binary
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create-release.outputs.upload_url }}
+          asset_path: ./ubuntu-release-archive/${{ steps.detect-archives.outputs.ubuntu-archive }}
+          asset_name: ${{ steps.detect-archives.outputs.ubuntu-archive }}
+          asset_content_type: application/gzip
+
+      - name: Upload macos release binary
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create-release.outputs.upload_url }}
+          asset_path: ./macos-release-archive/${{ steps.detect-archives.outputs.macos-archive }}
+          asset_name: ${{ steps.detect-archives.outputs.macos-archive }}
+          asset_content_type: application/gzip
+
+  merge_release_into_dev:
```

Comment:
Shall we automate this as well to ensure the timeliness?
I see two ways: 
1) comment `bors r+` right after you created the PR
2) just merge it with another action :)

What do you think?

Reply:
Yes, we can add mergify rules for that at some point :)